src/Build/Evaluation/ItemSpec.cs

```diff
@@ -277,32 +429,34 @@ private FileSpecMatcherTester FileMatcher
             }
         }
 
-        private IMSBuildGlob _msbuildGlob;
-
         // not a Lazy to reduce memory
-        protected virtual IMSBuildGlob MsBuildGlob
+        protected virtual IMSBuildGlob MsBuildGlob => _msbuildGlob ??= CreateMsBuildGlob();
+
+        protected ItemSpecFragment(string textFragment, string projectDirectory)
         {
-            get
-            {
-                if (_msbuildGlob == null)
-                {
-                    _msbuildGlob = CreateMsBuildGlob();
-                }
+            TextFragment = textFragment;
+            ProjectDirectory = projectDirectory;
+        }
 
-                return _msbuildGlob;
-            }
+        /// <returns>The number of times the
+        ///     <param name="itemToMatch"></param>
+        ///     appears in this fragment
+        /// </returns>
+        public virtual int MatchCount(string itemToMatch)
+        {
+            return IsMatch(itemToMatch)
+                ? 1
+                : 0;
```

Comment:
But that would mess up my Resharper formatting rules. I'd rather we adopt a formatting style for the entire repo and just apply that one on the entire code base, to avoid having to talk about formatting in each PR. You can create a separate issue for that.